### PR TITLE
Sort deployments and jobs in application API

### PR DESCRIPTION
**Pull request by mpolden**

### CODE REVIEW

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/ApplicationApiHandler.java [hunk 1]
```diff
@@ -347,9 +359,18 @@ else if (((Change.ApplicationChange)application.deploying().get()).revision().is
         for (URI rotation : rotations)
             globalRotationsArray.addString(rotation.toString());
 
-        // Deployments
+        // Deployments sorted according to deployment spec
```
Comment (bratseth) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/ApplicationApiHandler.java]: This is a lot of logic to add in ApplicationApiHandler.
I suggest moving it into the model - how about just let jobstatus() always return the jobs sorted?

**Review by bratseth — COMMENTED** [review]
Otherwise LGTM

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/ApplicationApiHandler.java (hunk 1 above)
Comment (mpolden) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/ApplicationApiHandler.java]: Agreed, will change.

Comment (mpolden) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/ApplicationApiHandler.java]: Right, I actually thought about doing that, but the problem is that you need access to deployment spec to sort and it's not available in `DeploymentJobs`.

Passing `DeploymentSpec` down to `DeploymentJobs` and storing a reference to it there just for the sake of sorting seems wrong. Suggestions?

Comment (bratseth) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/ApplicationApiHandler.java]: DeploymentJobs is owned by an Application, and cannot exist outside of an Application. You just need to pass the owning application to DeploymentJobs when it is created.
Then you can do owner.deploymentSpec() to access it.

**Comment by bratseth** [comment]
Also, note that we also have DeploymentOrder - may be possible to unify things a bit here.

**Comment by mpolden** [comment]
> You just need to pass the owning application to DeploymentJobs when it is created.
Then you can do owner.deploymentSpec() to access it.

DeploymentJobs is constructed when deserializing and before the Application instance is created. You could pass DeploymentSpec to DeploymentJobs at this time, but:

You also need to know the system to be able to map a step to a job and system is only known at runtime, which neither the application nor serializer currently knows. Changing all this just because of _one_ use case seems backwards.

> Also, note that we also have DeploymentOrder - may be possible to unify things a bit here.

True, I think reusing DeploymentOrder might be the best option.

**Comment by mpolden** [comment]
@bratseth PTAL

**Comment by bratseth** [comment]
Looks good! Merge?

**Pull request merged**